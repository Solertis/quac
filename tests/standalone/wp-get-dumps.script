#!/bin/bash

# Test that wp-get-dumps script works. To do so, we download dumps for one of
# the smallest Wikipedias, Afar (http://aa.wikipedia.org).
#
# This is a little risky, as mirror could stop working, the dump content could
# change, etc. But, we'll try it out. In this particular case, the wiki is
# "locked" as of 6/3/2013 (because there's no actual content), so that should
# help.
#
# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

. ./environment.sh

cd $DATADIR


mkdir wp-dumps
cat > quacrc <<EOF
[wkpd]
dump_dir = wp-dumps
bandwidth_limit = 10000
projects = aawiki
dumps = stub-meta-history.xml.gz pagelinks.sql.gz
EOF

# Do the mirror. The head pipe is to remove the statistics, which change; the
# "updating Wikimedia dumps" line also changes b/c it contains $DESTDIR.
y "wp-get-dumps --notimes --config quacrc 2>&1 | head -n -15 | fgrep -v 'updating Wikimedia dumps'"

x ls -Rhs .
