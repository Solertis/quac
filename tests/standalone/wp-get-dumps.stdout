$ (wp-get-dumps --notimes --config quacrc 2>&1 | head -n -15 | fgrep -v 'updating Wikimedia dumps')
wpdmp INFO     bandwidth limit is 10000 KB/s
receiving incremental file list
./
aawiki/
aawiki/latest/
aawiki/latest/aawiki-latest-pagelinks.sql.gz
aawiki/latest/aawiki-latest-stub-meta-history.xml.gz

$ ls -Rhs .
.:
total 4.0K
4.0K quacrc
   0 wp-dumps

./wp-dumps:
total 0
0 aawiki

./wp-dumps/aawiki:
total 0
0 latest

./wp-dumps/aawiki/latest:
total 84K
8.0K aawiki-latest-pagelinks.sql.gz
 76K aawiki-latest-stub-meta-history.xml.gz
